monasca-log-api/Dockerfile

```diff
@@ -0,0 +1,58 @@
+FROM python:2-alpine
+
+ARG LOG_API_REPO=https://github.com/openstack/monasca-log-api.git
+ARG LOG_API_BRANCH=master
+ARG CONSTRAINTS_BRANCH=master
+ARG CONSTRAINTS_FILE=https://git.openstack.org/cgit/openstack/requirements/plain/upper-constraints.txt?h=$CONSTRAINTS_BRANCH
+
+# To force a rebuild, pass --build-arg REBUILD="$(DATE)" when running
+# `docker build`
+ARG REBUILD=1
+
+ENV CONFIG_TEMPLATE=true \
+  LOG_LEVEL_ROOT=WARN \
+  LOG_LEVEL_CONSOLE=INFO \
+  LOG_LEVEL_ACCESS=INFO \
+  LOG_API_PORT=5607 \
+  KAFKA_URI=kafka:9092 \
+  KAFKA_WAIT_FOR_TOPICS=log \
+  KEYSTONE_IDENTITY_URI=http://keystone:35357 \
+  KEYSTONE_AUTH_URI=http://keystone:5000 \
+  KEYSTONE_ADMIN_USER=admin \
+  KEYSTONE_ADMIN_PASSWORD=secretadmin \
+  KEYSTONE_ADMIN_TENANT=admin \
+  KEYSTONE_ADMIN_DOMAIN=default \
+  ADD_ACCESS_LOG=true \
+  ACCESS_LOG_FORMAT="%(asctime)s [%(process)d] gunicorn.access [%(levelname)s] %(message)s" \
+  ACCESS_LOG_FIELDS='%(h)s %(l)s %(u)s %(t)s %(r)s %(s)s %(b)s "%(f)s" "%(a)s" %(L)s'
+
+ARG REBUILD_DEPENDENCIES=1
+RUN apk add --no-cache curl && \
+  apk add --no-cache --virtual build-dep git make g++ linux-headers && \
+  pip install --no-cache-dir Jinja2 gunicorn -c $CONSTRAINTS_FILE
+
+ARG REBUILD_CHECKOUT=1
+RUN git clone $LOG_API_REPO --depth 1 --branch $LOG_API_BRANCH monasca-log-api && \
```

Comment:
We've been using a [four-step clone](https://github.com/monasca/monasca-docker/blob/82c7e17fbbacd7c5ec079e15e211d94c2c2dcd3e/monasca-agent-base/Dockerfile#L28) in most of our other Dockerfiles since it gives more options for `..._BRANCH`. For instance here it would let us pass `--build-arg LOG_API_BRANCH=refs/changes/43/485443/2` to build directly from [an OpenStack patch](https://review.openstack.org/#/c/485443/).